website/demos/run_demos.py

```diff
@@ -22,19 +22,34 @@
 def launch_demo(demo_folder):
     subprocess.check_call([f"cd {demo_folder} && python run2.py"], shell=True)
 
+
+component_launch_code = """import gradio as gr
+with gr.Blocks() as demo:
+     component = gr.{component_name}()
+demo.launch()"""
 start_launch_time = time.time()
 for demo_name, port in demo_port_sets:
     demo_folder = os.path.join(GRADIO_DEMO_DIR, demo_name)
     demo_file = os.path.join(demo_folder, "run.py")
     demo_2_file = os.path.join(demo_folder, "run2.py")
-    with open(demo_file, "r") as file:
-        filedata = file.read()
-    assert "demo.launch()" in filedata, demo_name + " has no demo.launch()\n" + filedata 
+    if demo_name.endswith("_component"):
+        if os.path.exists(demo_file):
+            demo_file = os.path.join(demo_folder, "run.py")
+            with open(demo_file, "r") as file:
+                filedata = file.read()
+            assert "demo.launch()" in filedata, demo_name + " has no demo.launch()\n" + filedata
+        else:
+            os.mkdir(demo_folder)
+            filedata = component_launch_code.format(component_name=demo_name[:-10])
```

Comment:
I think it would be better if we defined a `SUFFIX="_component"` constant and then used it where it's needed.

Instead of `demo_name.endwith("_component")` we can do `demo_name.endswith(SUFFIX)` and `.format(component_name=demo_name[:-len(SUFFIX)]` as opposed to `format(component_name=demo_name[:-10])`

Reply:
Agreed, thanks.

Reply:
What will happen if we create a new component that requires constructor parameters? The demo will not work properly for sure, but will it break the website build?

Reply:
To be clear, there isn't a component right now with required parameters. For example, `CheckboxGroup` should get `choices` but they're not required in the constructor, they default to `None`

So

```python
import gradio as gr
with gr.Blocks() as demo:
     component = gr.CheckboxGroup()
demo.launch()
```

doesn't break the website, it just shows up like this:

<img width="1292" alt="Screen Shot 2022-07-08 at 5 50 52 PM" src="https://user-images.githubusercontent.com/9021060/178027415-f14c9898-94f6-4a7d-8eec-5b36003e14b2.png">

If we create a component with required parameters in the constructor, but don't override with a `_component` demo, it will break the website build.

Reply:
Ok got it, I think we should try catch this to be more robust but otherwise LGTM!